tools/depcheck/pkg/cmd/trace/filter_test.go

```diff
@@ -0,0 +1,247 @@
+package trace
+
+import (
+	"testing"
+
+	depgraph "github.com/openshift/origin/tools/depcheck/pkg/graph"
+)
+
+var vendorPkgs = &PackageList{
+	Packages: []Package{
+		{
+			Dir:        "/path/to/github.com/test/repo/vendor/github.com/testvendor/vendor_one",
+			ImportPath: "github.com/test/repo/vendor/github.com/testvendor/vendor_one",
+			Imports: []string{
+				"github.com/test/repo/vendor/github.com/testvendor/vendor_one/misc",
+			},
+		},
+		{
+			Dir:        "/path/to/github.com/test/repo/vendor/github.com/testvendor/vendor_one/misc",
+			ImportPath: "github.com/test/repo/vendor/github.com/testvendor/vendor_one/misc",
+			Imports: []string{
+				"github.com/test/repo/vendor/github.com/testvendor/vendor_three/vendor_three_misc",
+			},
+		},
+		{
+			Dir:        "/path/to/github.com/test/repo/vendor/github.com/testvendor/vendor_three",
+			ImportPath: "github.com/test/repo/vendor/github.com/testvendor/vendor_three",
+			Imports: []string{
+				"github.com/test/repo/vendor/github.com/testvendor/vendor_three/vendor_three_misc",
+			},
+		},
+		{
+			Dir:        "/path/to/github.com/test/repo/vendor/github.com/testvendor/vendor_three/vendor_three_misc",
+			ImportPath: "github.com/test/repo/vendor/github.com/testvendor/vendor_three/vendor_three_misc",
+			Imports: []string{
+				"fmt",
+			},
+		},
+		{
+			Dir:        "/path/to/github.com/test/repo/vendor/github.com/docker/docker-test-util",
+			ImportPath: "github.com/test/repo/vendor/github.com/docker/docker-test-util",
+			Imports: []string{
+				"log",
+				"github.com/test/repo/vendor/github.com/docker/docker-test-util/api",
+			},
+		},
+		{
+			Dir:        "/path/to/github.com/test/repo/vendor/github.com/docker/docker-test-util/api",
+			ImportPath: "github.com/test/repo/vendor/github.com/docker/docker-test-util/api",
+			Imports: []string{
+				"net",
+				"encoding/json",
+				"github.com/test/repo/vendor/github.com/google/glog",
+			},
+		},
+		{
+			Dir:        "/path/to/github.com/test/repo/vendor/github.com/google/glog",
+			ImportPath: "github.com/test/repo/vendor/github.com/google/glog",
+			Imports: []string{
+				"log",
+				"fmt",
+				"io",
+				"io/util",
+			},
+		},
+	},
+}
+
+func TestVendorPackagesCollapsedIntoRepo(t *testing.T) {
+	g, err := BuildGraph(vendorPkgs, []string{})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	vendorRoots := []string{
+		"github.com/test/repo/vendor/github.com/testvendor/vendor_one",
+		"github.com/test/repo/vendor/github.com/testvendor/vendor_three",
+		"github.com/test/repo/vendor/github.com/google/glog",
+		"github.com/test/repo/vendor/github.com/docker/docker-test-util",
+	}
+
+	filteredGraph, err := FilterVendorPackages(g, vendorRoots)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// should only have vendorRoots after being filtered
```

Comment:
not true. Add a node not under any of your vendor roots and update your test

Reply:
I test non-vendor nodes in the test below